packages/@aws-cdk/aws-codepipeline-actions/test/codestar-connections/codestar-connections-source-action.test.ts

```diff
@@ -148,26 +148,140 @@ describe('CodeStar Connections source Action', () => {
+              'Configuration': {
+                'EnvironmentVariables': '[{"name":"CommitId","type":"PLAINTEXT","value":"#{kornicameister.CommitId}"}]',
+              },
+            },
+          ],
+        },
+      ],
+    });
+  });
+
+  test('fail if variable from unused action is referenced', () => {
+    const stack = new Stack();
+    const pipeline = createBitBucketAndCodeBuildPipeline(stack, {});
+
+    const unusedSourceOutput = new codepipeline.Artifact();
+    const unusedSourceAction = new cpactions.CodeStarConnectionsSourceAction({
+      actionName: 'UnusedBitBucket',
+      owner: 'aws',
+      repo: 'aws-cdk',
+      output: unusedSourceOutput,
+      connectionArn: 'arn:aws:codestar-connections:us-east-1:123456789012:connection/12345678-abcd-12ab-34cdef5678gh',
+    });
+    const unusedBuildAction = new cpactions.CodeBuildAction({
+      actionName: 'UnusedCodeBuild',
+      project: new codebuild.PipelineProject(stack, 'UnusedMyProject'),
+      input: unusedSourceOutput,
+      outputs: [new codepipeline.Artifact()],
+      environmentVariables: {
+        CommitId: { value: unusedSourceAction.variables.commitId },
+      },
+    });
+    pipeline.stage('Build').addAction(unusedBuildAction);
+
+    expect(() => {
+      SynthUtils.synthesize(stack);
+    }).toThrow(/Cannot reference variables of action 'UnusedBitBucket', as that action was never added to a pipeline/);
+
+  });
+
+  test('fail if variable from unused action with custom namespace is referenced', () => {
+    const stack = new Stack();
+    const pipeline = createBitBucketAndCodeBuildPipeline(stack, {
+      variablesNamespace: 'kornicameister',
+    });
+    const unusedSourceOutput = new codepipeline.Artifact();
+    const unusedSourceAction = new cpactions.CodeStarConnectionsSourceAction({
+      actionName: 'UnusedBitBucket',
+      owner: 'aws',
+      repo: 'aws-cdk',
+      output: unusedSourceOutput,
+      connectionArn: 'arn:aws:codestar-connections:us-east-1:123456789012:connection/12345678-abcd-12ab-34cdef5678gh',
+      variablesNamespace: 'kornicameister',
+    });
+    const unusedBuildAction = new cpactions.CodeBuildAction({
+      actionName: 'UnusedCodeBuild',
+      project: new codebuild.PipelineProject(stack, 'UnusedProject'),
+      input: unusedSourceOutput,
+      outputs: [new codepipeline.Artifact()],
+      environmentVariables: {
+        CommitId: { value: unusedSourceAction.variables.commitId },
+      },
+    });
+    pipeline.stage('Build').addAction(unusedBuildAction);
+
+    expect(() => {
+      SynthUtils.synthesize(stack);
+    }).toThrow(/Cannot reference variables of action 'UnusedBitBucket', as that action was never added to a pipeline/);
   });
 });
 
-function createBitBucketAndCodeBuildPipeline(stack: Stack, props: Partial<cpactions.BitBucketSourceActionProps>): void {
+function createBitBucketAndCodeBuildPipeline(stack: Stack, props: Partial<cpactions.BitBucketSourceActionProps>): codepipeline.Pipeline {
```

Comment:
1. Let's make the second argument optional.
2. This actually uses the wrong (deprecated) interface.

```suggestion
function createBitBucketAndCodeBuildPipeline(stack: Stack, props: Partial<cpactions.CodeStarConnectionsSourceActionProps> = {}): codepipeline.Pipeline {
```